### ISSUE TITLE
Boolean precedence is wrong / Errors

### ISSUE THREAD

**Issue by bwrsandman** [glenfletcher/Equation#33]
The result of all these cases should always be False.

Current behaviour:

``` python
>>> import Equation
>>> Equation.Expression("a != a && c")(a=1, c=True)
False
>>> Equation.Expression("a != a && c")(a=2, c=True)  # ERROR
True
>>> Equation.Expression("a != a && c")(a=3, c=True)  # ERROR
True
>>> Equation.Expression("(a != a) && c")(a=1, c=True)
False
>>> Equation.Expression("(a != a) && c")(a=2, c=True)
False
>>> Equation.Expression("(a != a) && c")(a=3, c=True)
False
>>> 1 != 1 and True
False
>>> 2 != 2 and True
False
>>> 3 != 3 and True
False
```

It would seem that `&&` is evaluated before `!=`

**Comment by glenfletcher** [comment]
Fixed by Bug-Fix#33

**Comment by bwrsandman** [comment]
Thanks @glenflet
